dtdl-tools/src/lib.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+// This lib.rs is for running the .NET unit tests in this digital_twins_connector folder.
+// Running 'cargo test' will run all the .NET unit tests and the Rust unit tests.
+#[cfg(test)]
+mod digital_twins_connector_dotnet_tests {
+    use std::io::{self, Write};
+    use std::path::Path;
+    use std::process::Command;
+
+    // The manifest directory is the directory that contains the Cargo.toml file for this crate.
+    const MANIFEST_DIR: &str = env!("CARGO_MANIFEST_DIR");
+
+    const DTDL_VALIDATOR_FILENAME: &str = "dtdl-validator";
+    const DTDL_VALIDATOR_BIN_DIR: &str = "target/debug/dtdl-validator/bin/Debug/net7.0";
+    const DTDL_VALIDATOR_DIR_OPTION: &str = "-d";
+    const DTDL_VALIDATOR_EXT_OPTION: &str = "-e";
+
+    /// Validate DTDL files.
+    ///
+    /// # Arguments
+    /// * `directory` - The directory that contains the DTDL files that you wish to validate.
+    /// * `extension` - The file extension that the DTDL files use.
+    fn validate_dtdl_files(directory: &str, extension: &str) -> bool {
+        let dtdl_validator_command_path = Path::new(MANIFEST_DIR)
```

Comment:
users can customize the cargo target directory so this is not guaranteed to be the correct path. use `CARGO_TARGET_DIR` to get the path to the correct `target` folder. note that this just takes you to `target`, not `target/debug` or `target/release`

Reply:
I tried CARGO_TARGET_DIR , but it was not set at the time (no value).

BTW: Jordan used OUT_DIR, which is not ideal either.

Reply:
Annoyingly `OUT_DIR` is what cargo expects us to use for build script output as documented [here](https://doc.rust-lang.org/cargo/reference/build-scripts.html#outputs-of-the-build-script). There's nothing stopping us from violating these guidelines but I'd rather we try to align with them.

On a related note since `CARGO_TARGET_DIR` doesn't work, you may have to check for its existence and use a default otherwise, similar to what's mentioned [here](https://doc.rust-lang.org/cargo/reference/config.html#buildtarget-dir). You may also be able to set your own target directory as specified in that doc, though this does not solve the issue of detecting the profile